### ISSUE TITLE
Add functions to open, and close GRIB2 files

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/NCEPLIBS-g2c#276]
Given that this is a GRIB2 library, I would have expected to find a function that opens a GRIB2 file and reads the metadata in the file.

Yet there does not seem to be such a function.

Am I missing something?

@EricEngle-NOAA how do you open a file with the python library?

**Comment by edwardhartnett** [comment]
OK, I see that @EricEngle-NOAA answers my question in his documentation:

> GRIB2 file IO is performed directly in Python.  The unpacking/packing of GRIB2 integer, coded metadata and data sections is performed 
> by the g2c library functions via the g2clib Cython wrapper module.  The decoding/encoding of GRIB2 metadata is translated into more 
> descriptive, plain language metadata by looking up the integer code values against the appropriate GRIB2 code tables.  These code tables 
> are a part of the grib2io module.
>

**Comment by edwardhartnett** [comment]
We need functions in the library to open, create, and close GRIB2 files.

For example, I need such functions to test >2gb capabilities for NOAA-EMC/NCEPLIBS-g2c#43 and NOAA-EMC/NCEPLIBS-grib_util#294.

However, such functions would be very useful for many other tests as well. And also to users who otherwise have to implement all the file I/O.

**Comment by EricEngle-NOAA** [comment]
FWIW, I agree that the need is there for these functions.  I *think*, why the functionality is not there, is that g2c initially was just a port to C from the parent Fortran-based g2 library using f2c.  The g2 library depends on bacio (Byte-Addressable C IO).

**Comment by edwardhartnett** [comment]
Yes, and bacio, when I refactored it, came down to one C function that does file IO in a perfectly ordinary way. ;-)

I am changing hte g2 Fortran library so that it calls the C library, and all functionality will live in the C library. I've already started this with pngpack/jpegpack because they used copies of some old C code in the g2 library. Moving forward, the C library will provide pgppack/jpegpack and the Fortran library just calls the C library.

THese new file functions will get added after I release version 1.7.0, which is happening now. (See NOAA-EMC/NCEPLIBS-g2c#232 ).
